### PR TITLE
Cleanup kubernetes model modules #3774

### LINKED ISSUE #3774: [Cleanup] Use `osgi.include.resources.default` property in kubernetes model modules
### Is your task related to a problem? Please describe

While fixing https://github.com/fabric8io/kubernetes-client/issues/3761 we had created a property `osgi.include.resources.default` in #3771 which should contain common resource files to include when module jar is being packaged by maven bundle plugin (like `jandex.idx`). This property is used in all extension modules.

However, there are still some other places in this repository where these common resource files are duplicated again and again in maven bundle plugin configuration. I see it in all kubernetes model generator modules. You can see it with this grep command:
```
$ git grep -n "META-INF/jandex.idx"
``` 

### Describe the solution you'd like

- [X] Once #3771 gets merged into master.

We should refactor these duplicate occurences by using the property. For example here:

https://github.com/fabric8io/kubernetes-client/blob/a5c1d16da355acbf683727304e7e5308c3f1600b/kubernetes-model-generator/kubernetes-model-core/pom.xml#L148-L152

We should be using the property instead of specifying files again:
```diff
diff --git a/kubernetes-model-generator/kubernetes-model-core/pom.xml b/kubernetes-model-generator/kubernetes-model-core/pom.xml
index 35ec7accc..bcedbe16f 100644
--- a/kubernetes-model-generator/kubernetes-model-core/pom.xml
+++ b/kubernetes-model-generator/kubernetes-model-core/pom.xml
@@ -146,8 +146,7 @@
               io.fabric8.kubernetes.api.model**,
             </Export-Package>
             <Include-Resource>
-              {maven-resources},
-              /META-INF/jandex.idx=target/classes/META-INF/jandex.idx,
+              ${osgi.include.resources.default},
               /core.properties=target/classes/core.properties,
             </Include-Resource>
           </instructions>
```

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by anurag-rajawat**
## Description
Fix #3774 

## Type of change

 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [ ] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [x] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [x] Code contributed by me aligns with current project license: [Apache 2.0](https://www.apache.org/licenses/LICENSE-2.0)
 - [ ] I Added [CHANGELOG](../CHANGELOG.md) entry regarding this change
 - [ ] I have implemented unit tests to cover my changes
 - [ ] I have added/updated the [javadocs](https://www.javadoc.io/doc/io.fabric8/kubernetes-client/latest/index.html) and other [documentation](https://github.com/fabric8io/kubernetes-client/blob/master/doc/CHEATSHEET.md) accordingly
 - [x] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=fabric8io_kubernetes-client) report
 - [ ] I tested my code in Kubernetes
 - [ ] I tested my code in OpenShift

### CODE REVIEW

**Comment by anurag-rajawat** [comment]
@rohanKanojia why these 3 tests failed?

**Comment by anurag-rajawat** [comment]
@rohanKanojia is there anything to do?

**Comment by rohanKanojia** [comment]
@anurag-rajawat :those failures were due to some flaky tests. I restarted them and now they're passing.

You just need to wait until your PR gets reviewed. We don't work on weekends.

File: kubernetes-model-generator/kubernetes-model-admissionregistration/pom.xml [hunk 1]
```diff
@@ -32,6 +32,83 @@
 
   <properties>
     <clone-kube>true</clone-kube>
+    <osgi.include.resources>
+      ${osgi.include.resources.default},
+      /core.properties=target/classes/core.properties,
```
Comment (rohanKanojia) [inline kubernetes-model-generator/kubernetes-model-admissionregistration/pom.xml:37]: This doesn't look right. It should be the same as the values provided in `<Include-Resource>` plugin configuration which you deleted below.

File: kubernetes-model-generator/kubernetes-model-admissionregistration/pom.xml [hunk 2]
```diff
@@ -32,6 +32,83 @@
...
+    </osgi.include.resources>
+    <osgi.import>
+      *,
+      io.fabric8.kubernetes.api.model.admission.v1,
```
Comment (rohanKanojia) [inline kubernetes-model-generator/kubernetes-model-admissionregistration/pom.xml:41]: This should be the same as `<Import-Package>` plugin config below.

File: kubernetes-model-generator/kubernetes-model-admissionregistration/pom.xml [hunk 3]
```diff
@@ -32,6 +32,83 @@
...
+      io.fabric8.openshift.api.model.miscellaneous.cloudcredential.v1;resolution:=optional,
+      io.fabric8.openshift.api.model.miscellaneous.cncf.cni.v1;resolution:=optional,
+      io.fabric8.openshift.api.model.miscellaneous.imageregistry.operator.v1;resolution:=optional,
+      io.fabric8.openshift.api.model.miscellaneous.metal3.v1alpha1;resolution:=optional,
+      io.fabric8.openshift.api.model.miscellaneous.network.operator.v1;resolution:=optional,
+    </osgi.import>
+    <osgi.export>
+      io.fabric8.kubernetes.api,
+      io.fabric8.kubernetes.api.builder,
+      io.fabric8.kubernetes.api.model**,
```
Comment (rohanKanojia) [inline kubernetes-model-generator/kubernetes-model-admissionregistration/pom.xml:107]: This should b same as `<Export-Package>` plugin configuration you deleted below

**Comment by rohanKanojia** [comment]
Looks like you just copy pasted my suggestion into every module. You need to change it in every module with respect to provided plugin configuration.

Also, please make sure project is building fine using `mvn clean install` after doing change in every module

File: kubernetes-model-generator/kubernetes-model-admissionregistration/pom.xml (hunk 3 above)
Comment (anurag-rajawat) [reply kubernetes-model-generator/kubernetes-model-admissionregistration/pom.xml]: Done

**Comment by anurag-rajawat** [comment]
@rohanKanojia I've again build the project with both `jdk-8` and `jdk-11` and it has been successful in both builds, but here it got failed, why?

Also, I'm attaching screenshot for your reference

With `jdk-8`
Screenshot from 2022-02-09 16-25-15

With `jdk-11`
Screenshot from 2022-02-09 17-16-26

**Comment by rohanKanojia** [comment]
It seems to be a flaky failure. Let me restart the job

**Review by rohanKanojia — APPROVED** [review]

**Comment by anurag-rajawat** [comment]
@manusa please review this PR

**Review by manusa — APPROVED** [review]
LGTM, thx!

**Pull request merged**